### ISSUE TITLE
CVE-2021-23386: out-dated dns-packer dependency

### ISSUE THREAD

**Issue by arderyp** [symfony/webpack-encore#986]
[CVE-2021-23386](https://nvd.nist.gov/vuln/detail/CVE-2021-23386)

```
 yarn why dns-packet
yarn why v1.22.5
[1/4] Why do we have the module "dns-packet"...?
[2/4] Initialising dependency graph...
[3/4] Finding dependency...
[4/4] Calculating file sizes...
=> Found "dns-packet@1.3.1"
info Reasons this module exists
   - "@symfony#webpack-encore#webpack-dev-server#bonjour#multicast-dns" depends on it
   - Hoisted from "@symfony#webpack-encore#webpack-dev-server#bonjour#multicast-dns#dns-packet"
info Disk size without dependencies: "96KB"
info Disk size with unique dependencies: "204KB"
info Disk size with transitive dependencies: "204KB"
info Number of shared dependencies: 2
Done in 0.60s.
```

RELATED:
- https://github.com/watson/bonjour/issues/63
- https://github.com/mafintosh/multicast-dns/issues/74

**Comment by arderyp** [comment]
`bonjour` may be the sticking point.  It looks like it hasn't been touched in 5 years.

**Comment by weaverryan** [comment]
Hi there!

This type of things needs to be fixed "downstream". Or, to say it differently, if/when there is a fix downstream, there is nothing preventing you from getting it (as Encore doesn't rely directly on the package in question). If a fix for that library DOES come out, and it requires some major dependency upgrade in Encore to allow it, then there is something we can do. If that happens, definitely please open a PR or issue :).

Additionally, from looking at the library and CVE, it seems unlike that this affects encore in any real way (but I realize that nobody wants packages with security vulnerabilities in their chain, regardless)

Cheers!

**Comment by pavlakis** [comment]
As mentioned above, this is not directly related to this package.

However, in order to resolve, you can follow these steps:

 yarn remove @symfony/webpack-encore
 yarn add --dev multicast-dns
 yarn add --dev @symfony/webpack-encore
 
 And also add the following to package.json as per https://github.com/webpack/webpack-dev-server/issues/3340#issuecomment-847659624
 
 ```json
     "resolutions": {
         "multicast-dns": "7.2.3"
     },
 ```

**Comment by arderyp** [comment]
fair enough @weaverryan, and thanks for the suggestion @pavlakis

**Comment by arderyp** [comment]
the fix has been back ported to version `1.3.4`: https://github.com/watson/bonjour/issues/63#issuecomment-849060227

Still awaiting GH to update the advisory to consider this version valid and squash the vulnerability warning for those that update to it.

**Comment by arderyp** [comment]
`1.3.4` provides the fix and does not require downstream versioning.  A simple `yarn upgrade` should resolve the vulnerability now.
